### ISSUE TITLE
Simd types

### ISSUE THREAD

**Issue by lerno** [c3lang/c3c#56]

**Comment by lerno** [comment]
@data-man I think you looked at this for Zig?

**Comment by data-man** [comment]
I don't use Zig anymore. :)

> 3, 4

Are these matrices?

**Comment by lerno** [comment]
Not quite, more like multi dimensional arrays, that is - availability of matrix calculations are not guaranteed. However, it's likely that it is better to model this as structs. Since I haven't used SIMD at all in C – I haven't had the need in my applications – I'm very much unsure of the ideal implementation.

However, I note that a lot of additional functionality added by GCC/Clang appears mostly unused. Consequently I draw the conclusion that aside from the simd vector types in GCC/Clang with associated functions, there is no real consensus as to what could be done to simplify simd programming.

**Comment by lerno** [comment]
There is a very simple question, which is whether arbitrary definition of vector types should be allowed, even if they do not have a native register – this seems to be the way of GCC/Clang – or if it is better to provide the various native types with their own names, as an example we might consider:

```c
float[<4>] x;
// vs
floatx4 x;
```

**Comment by lerno** [comment]
For now we have the [<>] syntax for vectors. I am amenable to updating this syntax in favour of a better proposal, but for now...
